public/locations-list-map.js

```diff
@@ -770,34 +787,37 @@ function centerMapToMarkersNearCoords(latitude, longitude) {
       distance = distance + 1;
     }
 
-    markerDistances[distance] = marker;
-
-    distances.push(distance);
+    markerDistances.set(distance, marker);
   }
 
-  // sort the distances and set bounds to closest three
-  distances.sort((a, b) => a - b);
+  // order markerDistances by key (distance)
+  let distances = [...markerDistances.keys()].sort((a,b) => a -b);
+  // return array of markers in order of distance ascending
+  return distances.map((distance) => markerDistances.get(distance));
+}
+
+/**
+ * Centers map around markers nearest to an arbitrary set of latitude/longitude coordinates.
+ */
+function centerMapToMarkersNearCoords(latitude, longitude) {
```

Comment:
You should be able to get rid of this function and just use fitMapToMarkersNearBounds. In the "near me" case, you can construct a bounds with the user's location. The hasMarker logic should apply to fitMapToMarkersNearBounds too.